src/scripts/main.js

```diff
@@ -1,3 +1,45 @@
 'use strict';
 
-// write code here
+const thead = document.querySelector('thead');
+
+function convertToNumber(currency) {
+  const str = currency.slice(1);
+  let result = '';
+
+  for (let i = 0; i < str.length; i++) {
+    if (str[i] === ',') {
+      result += '.';
+    } else {
+      result += str[i];
+    }
+  }
+
+  return +result;
+}
+
+thead.addEventListener('click', (e) => {
+  const title = e.target.textContent;
+  const index = e.target.cellIndex;
+
+  const tbody = document.querySelector('tbody');
+  const rows = [...tbody.rows];
+
+  rows.sort((rowA, rowB) => {
+    const valueA = rowA.cells[index].textContent;
+    const valueB = rowB.cells[index].textContent;
+
+    switch (title) {
+      case 'Name':
+      case 'Position':
+        return valueA.localeCompare(valueB);
+
+      case 'Age':
+        return valueA - valueB;
+
+      case 'Salary':
+        return convertToNumber(valueA) - convertToNumber(valueB);
+    }
```

Comment:
In the `sort` function, you're accessing the `textContent` of each cell for every comparison. This could be a performance issue if there are many rows in the table. A more efficient approach would be to map the rows to an array of objects that contain both the original row and the values to sort by, then sort that array. This way, you only need to access the `textContent` once for each cell.